src/sql/workbench/services/connection/test/browser/connectionDialogWidget.test.ts

```diff
@@ -0,0 +1,175 @@
+		let providerDisplayNames = ['Mock SQL Server', 'Mock SQL Server 1'];
+		let providerNameToDisplayMap = { 'MSSQL': 'Mock SQL Server', 'PGSQL': 'Mock SQL Server 1' };
+		mockConnectionManagementService.setup(x => x.getUniqueConnectionProvidersByNameMap(TypeMoq.It.isAny())).returns(() => {
+			return getUniqueConnectionProvidersByNameMap(providerNameToDisplayMap);
+		});
+		connectionDialogWidget.updateConnectionProviders(providerDisplayNames, providerNameToDisplayMap);
+		assert.equal(connectionDialogWidget.getDisplayNameFromProviderName('PGSQL'), providerNameToDisplayMap['PGSQL']);
+	});
+
+	test('setting newConnectionParams test for connectionDialogWidget', () => {
+		let params: INewConnectionParams = {
+			connectionType: ConnectionType.editor,
+			input: {
+				onConnectReject: undefined,
+				onConnectStart: undefined,
+				onDisconnect: undefined,
+				onConnectSuccess: undefined,
+				onConnectCanceled: undefined,
+				uri: 'Editor Uri'
+			},
+			runQueryOnCompletion: RunQueryOnConnectionMode.executeQuery,
+			providers: ['MSSQL']
+		};
+		let providerNameToDisplayMap = { 'MSSQL': 'Mock SQL Server' };
+		mockConnectionManagementService.setup(x => x.getUniqueConnectionProvidersByNameMap(TypeMoq.It.isAny())).returns(() => {
+			return getUniqueConnectionProvidersByNameMap(providerNameToDisplayMap);
+		});
+		connectionDialogWidget.newConnectionParams = params;
+		assert.equal(connectionDialogWidget.newConnectionParams, params);
+	});
+
+	test('open should call onInitDialog', async () => {
+		let params: INewConnectionParams = {
+			connectionType: ConnectionType.editor,
+			input: {
+				onConnectReject: undefined,
+				onConnectStart: undefined,
+				onDisconnect: undefined,
+				onConnectSuccess: undefined,
+				onConnectCanceled: undefined,
+				uri: 'Editor Uri'
+			},
+			runQueryOnCompletion: RunQueryOnConnectionMode.executeQuery,
+			providers: ['MSSQL']
+		};
+		let providerNameToDisplayMap = { 'MSSQL': 'Mock SQL Server' };
+		mockConnectionManagementService.setup(x => x.getUniqueConnectionProvidersByNameMap(TypeMoq.It.isAny())).returns(() => {
+			return getUniqueConnectionProvidersByNameMap(providerNameToDisplayMap);
+		});
+		//params must be assigned to get load providers
+		connectionDialogWidget.newConnectionParams = params;
+		let mockConnectionProfile = createConnectionProfile('test_id', '');
+		mockConnectionManagementService.setup(x => x.getRecentConnections(TypeMoq.It.isValue(params.providers))).returns(() => {
+			return [mockConnectionProfile];
+		});
+		let called = false;
+		connectionDialogWidget.onInitDialog(() => {
+			called = true;
+		});
+		return connectionDialogWidget.open(true).then(() => {
+			called = false;
+			return connectionDialogWidget.open(false).then(() => {
+				assert(called);
+			});
+		});
+	});
+
+	test('get set tests for connectButtonState and databaseDropdownExpanded', () => {
+		connectionDialogWidget.connectButtonState = true;
```

Comment:
Again another equality check test?

Reply:
another test to check if the setter works, that's correct